srtcore/channel.h

```diff
@@ -264,30 +257,33 @@ class CChannel
 
         if (adr.family() == AF_INET)
         {
-            mh.msg_control = m_acCmsgSendBuffer;
-            mh.msg_controllen = CMSG_SPACE(sizeof(in_pktinfo));
-            cmsghdr* cmsg_send = CMSG_FIRSTHDR(&mh);
+            mh.msg_control = (void *) buf;
+            mh.msg_controllen = sizeof(struct in_pktinfo) + sizeof(cmsghdr);
```

Comment:
If you move the creation of the local variable to here, you can simply use `Size(buf)`.

Reply:
Provided you don't, though, why it doesn't use `CMSG_SPACE`?

Reply:
I followed some examples I found in [there](https://man.archlinux.org/man/core/man-pages/CMSG_FIRSTHDR.3.en) (CMSG_FIRSTHDR manpage). They are setting controllen to `msg.msg_controllen = sizeof(u.buf);`, where u.buf is the equivalent of the control buffer I'm using

Reply:
But this buffer is declared as:
```
    char buf[CMSG_SPACE(sizeof(myfds))];
```
so sizeof on the whole buffer could be treated as an alias in this case.

Reply:
At one point when I was trying to fix this issue I used 2 different buffers, one for IPv4 and another for IPv6. Both of them used CMSG_SPACE in their declaration. That's why I didn't use it there, but you are definitively right I'll fix that

Reply:
I did it this way initially, but then it turned out to be a bad idea. I found cases when at least theoretically the same CMSG structure may contain entries for both IPv4 and IPv6 pktinfo and then both are found.

See also the comment in `channel.h` line 194.

Reply:
Not using CMSG_SPACE caused also another issue that you asked me about in another comment. I added `sizeof(cmsghdr)` to avoid CMSG_DATA returning NULL. Thanks for your input, I'll push something that address all of it soon